core/blockchain.go

```diff
@@ -2131,6 +2133,40 @@ func (bc *BlockChain) WriteShardLastCrossLink(shardID uint32, cl types.CrossLink
 	return rawdb.WriteShardLastCrossLink(bc.db, cl.ShardID(), cl.Serialize())
 }
 
+// UpdateShardLastCrossLink saves the last crosslink of a shard
+func (bc *BlockChain) UpdateShardLastCrossLink(shardID uint32, cl types.CrossLink) error {
```

Comment:
please describe the logic in this func. Even by reading the code I don't fully get what's the meaning of this func.

Reply:
Then it shouldn't be called last right? It should be called "LastContinuousCrossLink"

Reply:
good name! changed